### DISCUSSION TITLE
Change isValid when form becomes valid in mode: onSubmit

### DISCUSSION THREAD

**Discussion by m-deacon** [react-hook-form/react-hook-form#2971] [Q&A] [answered]
Hi guys

I recently made the switch from formik to RHF and so far the experience has been pretty solid. I'm just struggling with one thing, I tried to search these discussions / issues for a similar issue, but couldn't find anything.

I have a login form that has an email and a password with a submit button. I have validation on the password field of length > 8 and I have the submit button disabled if `isValid` is false.

Is there a way to have `isValid` update to `true` when the password field has reached 8 characters (i.e the form becomes valid) but they haven't blurred or left the field. As a user I don't want to need to click outside the form before I can click submit (or pressing enter), but I don't want to be able to submit with invalid data. The only way this works is `mode: onChange`, but I don't want to show the error message when the user starts typing.

I hope this makes sense.

TIA

**Comment by bluebill1049** [comment]
i think `onTouched` mode is probably what you after.

**Comment by m-deacon** [comment, marked as answer]
`onTouched` requires the user to click outside of the input / doesn't allow them to click enter to submit after typing the 8 chars. I've tried all the available options and some other configs, but nothing seems to work. Am I missing something?

**Reply by bluebill1049** [reply]
i don't think you are missing anything, as right now isValid works with the mode: 'onChange', `onBlur` and `onTouched` if those options doesn't works for you, then you should probably consider mautally trigger validation.

**Reply by m-deacon** [reply]
In that case, I'd like to open the discussion for this to possibly be a built-in behavior or could someone assist in how I'd go about doing this through manually triggered validation? Though I do feel it's a common pattern for the form to not return any errors, but change to being valid when it becomes valid without the user having to blur the last input.

**Reply by bluebill1049** [reply]
i think you can easily do what you described above with onChagne mode and play around with the `formState` whether is touched, show error only the input is dirty.

**Reply by m-deacon** [reply]
Ahhhhhh - for some reason I had removed the touched handling in my input component. Yeah it's working as intended now 🤦‍♀️ 

Thanks @bluebill1049

**Reply by bluebill1049** [reply]
all good 👍
